contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1645,8 +1664,12 @@ check_alter_server_stmt(GrantRoleStmt *stmt)
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 				 errmsg("%s is not a login", grantee_name)));
 
-	/* only sysadmin role is assumed below */
-	if (!has_privs_of_role(GetSessionUserId(), sysadmin))
+	/* 
+	 * check if it has sysadmin privileges or
+	 * if server role is securityadmin and it has privileges of securityadmin
+	 */
+	if (!has_privs_of_role(GetSessionUserId(), sysadmin) && ((strcmp(granted_name, BABELFISH_SECURITYADMIN) != 0)
+										|| !has_privs_of_role(GetSessionUserId(), get_securityadmin_oid())))
```

Comment:
2nd condition is a bit confusing, can we refactor it a bit ?

Reply:
are we expecting to just give the negation of the second condition instead of expanding the negation?